src/libhirte/common/config.c

```diff
@@ -0,0 +1,180 @@
+
+        const char *option_names[] = { CFG_LOG_LEVEL, CFG_LOG_TARGET, CFG_LOG_IS_QUIET };
+
+        int length = sizeof(env_vars) / sizeof(env_vars[0]);
+        for (int i = 0; i < length; i++) {
+                char *value = getenv(env_vars[i]);
+                if (value != NULL && strlen(value) > 0) {
+                        // environment variable is defined and has a non-empty value, so store it to the configuration
+                        int result = config_set_value(option_names[i], value);
+                        if (result != 0) {
+                                return result;
+                        }
+                }
+        }
+        return 0;
+}
+
+int config_set_value(const char *option_name, const char *option_value) {
+        char *name_copy = strdup(option_name);
+        if (name_copy == NULL) {
+                return ENOMEM;
+        }
+
+        char *value_copy = (char *) option_value;
+        if (value_copy != NULL) {
+                if (strlen(value_copy) > 0) {
+                        // storing non NULL value, so copy is needed
+                        value_copy = strdup(option_value);
+                        if (value_copy == NULL) {
+                                free(name_copy);
+                                return ENOMEM;
+                        }
+                } else {
+                        // empty string mean non-existent value, so store NULL
+                        value_copy = NULL;
+                }
+        }
+
+        struct config_option *replaced = hashmap_set(
+                        config, &(struct config_option){ .name = name_copy, .value = value_copy });
+        if (hashmap_oom(config)) {
+                // OOM during hashmap set operation
+                free(name_copy);
+                free(value_copy);
+                return ENOMEM;
+        }
+
+        if (replaced != NULL) {
+                // option has been replaced with the new one, we need to deallocate the old one
+                free(replaced->name);
+                free(replaced->value);
+        }
+        return 0;
+}
+
+const char *config_get_value(const char *option_name) {
+        struct config_option *found = (struct config_option *) hashmap_get(
+                        config, &(struct config_option){ .name = (char *) option_name });
+        if (found != NULL) {
+                return found->value;
+        } else {
+                return NULL;
+        }
+}
+
+bool config_get_bool_value(const char *option_name) {
+        bool result = false;
+        const char *value = config_get_value(option_name);
+        if (value != NULL) {
+                result = strcasecmp("true", value) == 0 || strcasecmp("yes", value) == 0 ||
+                                strcasecmp("on", value) == 0;
+        }
+        return result;
+}
+
+void config_iterate(bool (*iter)(const void *item, void *udata), void *udata) {
+        hashmap_scan(config, iter, udata);
+}
```

Comment:
Make sure to add an empty newline at the end of the file

Reply:
Hmm, not sure why linter didn't scream on it :-(
Fixed